eng/pipelines/templates/steps/template-test-run-job.yml

```diff
@@ -12,15 +12,22 @@ steps:
       parameters:
           Version: $(AzdVersion)
 
-    - template: /eng/pipelines/templates/steps/azd-login.yml
-
-    # Java pre-requisites that isn't met
-    # TODO: Use azd container as a form of validation that the container works
-    - template: /eng/pipelines/templates/steps/install-ms-openjdk.yml
-
     # Required to clone repos that are not yet public
     - template: /eng/pipelines/templates/steps/set-git-credentials.yml
 
+    # Use the AzureCLI@2 task to set the variables azd will need to use later to authenticate over OIDC.
+    - task: AzureCLI@2
```

Comment:
I'll move this into its own step, so we can share it with what I'm doing for `build-cli`.